packages/inferno-component/src/index.ts

```diff
@@ -286,8 +285,8 @@ export default class Component<P, S> implements ComponentLifecycle<P, S> {
 			}
 
 			/* Update if scu is not defined, or it returns truthy value or force */
-			if (isUndefined(this.shouldComponentUpdate) || this.shouldComponentUpdate(nextProps, nextState, context) || force) {
-				if (!isUndefined(this.componentWillUpdate)) {
+			if (isNullOrUndef(this.shouldComponentUpdate) || (this.shouldComponentUpdate && this.shouldComponentUpdate(nextProps, nextState, context)) || force) {
```

Comment:
I think this should be `(isFunction(this.shouldComponentUpdate) && this.shouldComponentUpdate(nextProps, nextState, context))` where `isFunction` is imported from `inferno-shared`.